### ISSUE TITLE
Trouble building on MacOS Catalina 10.15.6 --> MACOSX_RPATH

### ISSUE THREAD

**Issue by Xalip** [eclipse-paho/paho.mqtt.c#935]
**Describe the bug**
seems like during compilation MACOSX_RPATH isn't set correctly

**To Reproduce**
1. `git clone https://github.com/eclipse/paho.mqtt.c.git`
2. `cd paho.mqtt.c`
3. `git checkout v1.3.1`
4. `cmake -Bbuild -H. -DPAHO_WITH_SSL=OFF -DPAHO_ENABLE_TESTING=OFF`
3.5 `sudo cmake --build build/ --target install`

If I try to build my project I'm getting following error: 
![image](https://user-images.githubusercontent.com/18552380/90035495-2756fd80-dcc2-11ea-962d-954a2b4834c6.png)

CMakeLists.txt:
```
cmake_minimum_required(VERSION 3.16)
project(aip_gateway)

set(CMAKE_CXX_STANDARD 14)

set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11 -pthread")

# search installed paho package
find_package(PahoMqttCpp CONFIG REQUIRED)

# make projektarbeit binary executable
add_executable(aip_gateway src/main.cpp src/incomeMessageHandler.cpp src/mqtt_client.cpp src/mqtt_message.cpp src/publishToBroker.cpp)

# link all necessary installed libs
target_link_libraries(aip_gateway PahoMqttC::PahoMqttC paho-mqttpp3 paho-mqtt3c paho-mqtt3a)
```

**Expected behavior**
during build all libs get correctly linked

** Environment (please complete the following information):**
 - OS: [e.g. MacOS Catalina]
 - Version [e.g. 10.15.6]

**Comment by icraggs** [comment]
The build message seems to indicate that the library paho-mqttpp3 can't be found.  That's not a name I use.  You could look at the Travis builds for the Mac and see how those are done, that might help.  For instance:

https://travis-ci.org/github/eclipse/paho.mqtt.c/jobs/717342837

Also, why use 1.3.1 rather than the most recent version, 1.3.5?

**Comment by Xalip** [comment]
I've just followed these instructions in their readme: https://github.com/eclipse/paho.mqtt.cpp, somewhere down they say to checkout the older branch. Anyway, now I've used the `travis-build.sh` and set some env vars before and everything compiled like a charm. Somehow during compiling my project it still fails due to a linking error. I don't think there is any relation to this lib. So thanks for your quick response.
